data/systemd/snapd.apparmor.service

```diff
@@ -0,0 +1,25 @@
+# This systemd unit is needed on distributions that use apparmor but don't have
+# special support for loading snapd apparmor profiles. Until upstream apparmor
+# user-space release contains a systemd unit that is actually shipped by
+# distributors and that contains the necessary extension points for snapd the
+# apparmor profiles for snap applications need to be loaded separately from
+# other applications.
+[Unit]
+Description=Load AppArmor profiles managed internally by snapd
+DefaultDependencies=no
+Before=sysinit.target
+Before=snapd.service
+# This dependency is meant to ensure that apparmor initialization (whatever that might entail) is complete.
+After=apparmor.service
+ConditionSecurity=apparmor
+
+[Service]
+Type=oneshot
+ExecStart=/bin/sh -c 'for prof in /var/lib/snapd/apparmor/profiles/*; do test "${prof%\~}" != "${prof}" && continue; apparmor_parser --replace --write-cache --cache-loc=/var/cache/apparmor -O no-expr-simplify --quiet "$prof"; done'
+ExecReload=/bin/sh -c 'for prof in /var/lib/snapd/apparmor/profiles/*; do test "${prof%\~}" != "${prof}" && continue; apparmor_parser --replace --write-cache --cache-loc=/var/cache/apparmor -O no-expr-simplify --quiet "$prof"; done'
```

Comment:
It might be nice to break these out into a script. Eg, /usr/lib/snapd/snap-apparmor.systemd, then do:
```
ExecStart=/usr/lib/snapd/snap-apparmor.systemd start
ExecReload=/usr/lib/snapd/snap-apparmor.systemd reload
ExecStop=/usr/lib/snapd/snap-apparmor.systemd stop
```
then adjust start/reload/stop in snap-apparmor.systemd as desired.